src/select.ts

```diff
@@ -193,14 +191,14 @@ export class SelectQuery<
     const table = fromItem as Table<any, any>;
 
     return this.newSelectQuery(
-      [...this.tokens, new StringToken(`FROM`), new TableToken(table)],
+      [...this.tokens, new StringToken(`FROM`), ...table.toTokens()],
       table,
     ) as any;
   }
 
   join<T extends Table<any, any>>(table: T): Join<SelectQuery<Columns, IncludesStar>, T> {
     return this.newSelectQuery(
-      [...this.tokens, new StringToken(`JOIN`), new TableToken(table)],
+      [...this.tokens, new StringToken(`JOIN`), ...(table as Table<any, any>).toTokens()],
```

Comment:
The type parameter `T` is already bounded  (`T extends Table<any, any>`). Do you still need the `as Table<any, any>`?

Reply:
Yeah this was weird.  Without the cast, it thinks table has the type: `table: T extends unknown`.  This may have to do with using `Table<any, any>`, and it isn't able to figure out the columns.  It works when we can ascribe it a type, such as when it previously was passed in to a constructor.

Reply:
I was curious so I poked around a bit.
1. Upgrading TypeScript from 4.x to 5.x didn't fix it. (Though probably worth doing that anyway in a separate PR.)
2. We talked about type ascription working but I realized that `Table<any, any>` expands to `any & InternalTable<any, any>`, so even this works:

```ts
const t: Table<any, any> = 1; // wat
```

You mentioned that doing `extends Table<string, unknown>` didn't quite work. But maybe it will work if we upgrade to TypeScript 5.x?
- I tried the upgrade but was thwarted by what appears to be a formatting difference in the snapshot tests. Maybe we can just save the new snapshots? https://github.com/Anrok/mammoth/pull/10

Reply:
Yeah upgrading and trying to get this to work makes sense to me.  I'll checkout your branch and see if I can get the tests to pass.

Reply:
I fixed this after integrating the type upgrade.  I also had to update the from/join types to allow a `FromItem | Table`.  I added a type-alias for this since it was used in many locations.  This change allows removing the cast.